backends/dpdk/dpdkArch.cpp

```diff
@@ -1095,6 +1095,106 @@ const IR::Node* PrependPDotToActionArgs::preorder(IR::MethodCallExpression* mce)
             arguments);
 }
 
+const IR::Node* DismantleMuxExpressions::preorder(IR::Mux* expression) {
+    // We always dismantle muxes for dpdk
+    auto type = typeMap->getType(getOriginal(), true);
+    visit(expression->e0);
+    auto tmp = createTemporary(type);
+    auto save = statements;
+    statements.clear();
+    visit(expression->e1);
+    (void) addAssignment(expression->srcInfo, tmp, expression->e1);
```

Comment:
Are you sure this does not need to be added to the typeMap?
What if you have an expression like a ? (b ? c : d) : (e ? f : g)